lua/sqlite/defs.lua

```diff
@@ -47,9 +47,33 @@ local clib = (function()
       end
 
       if os.sysname == "Darwin" then
-        return os.machine == "arm64" and "/opt/homebrew/opt/sqlite/lib/libsqlite3.dylib"
+        path = os.machine == "arm64" and "/opt/homebrew/opt/sqlite/lib/libsqlite3.dylib"
           or "/usr/local/opt/sqlite3/lib/libsqlite3.dylib"
       end
+
+      if file_exists(path) then
+        return path
+      end
+
+      -- check for nix installation
+      path = string.format("%s/.nix_profile/bin/sqlite3", luv.os_getenv "HOME")
+      if file_exists(path) then
+        return path
+      end
+
+      -- as a last resort, try using `which sqlite3`
+      local handle = io.popen "which sqlite3"
```

Comment:
Hmm, it seemed to work for me. I'll do some more investigation today.

Reply:
It looks like there is actually not really a way to use `which sqlite3` to get the `dylib`. It looks like this worked for me on `nix` because the linked binary has the `dylib` statically in the binary or something. I'll close the PR.